controller/model/edge_service_manager.go

```diff
@@ -259,14 +250,8 @@ func (result *ServiceListResult) collect(tx *bbolt.Tx, ids []string, queryMetaDa
 	identityServiceConfigs := result.manager.env.GetStores().Identity.LoadServiceConfigsByServiceAndType(tx, result.identityId, result.configTypes)
 
 	for _, key := range ids {
-		if !result.isAdmin && result.identityId != "" {
-			service, err = result.manager.ReadForNonAdminIdentityInTx(tx, key, result.identityId)
-		} else {
-			service, err = result.manager.readInTx(tx, key)
-			if service != nil && result.isAdmin {
-				service.Permissions = []string{db.PolicyTypeBindName, db.PolicyTypeDialName}
-			}
-		}
+		// service permissions for admin & non-admin identities will be set according to policies
+		service, err = result.manager.ReadForIdentityInTx(tx, key, result.identityId, result.configTypes)
```

Comment:
☝️ this will now happen when `result.identityId` == "", and I'm not really sure what that means or if it's ok.

Reply:
I'm not sure it can happen, but we could short-circuit it in queryServices. If the `identityId` is `""` we could return the service list result before we run the query.